protocol/src/lib.rs

```diff
@@ -201,9 +199,23 @@ impl Default for EnvelopeParams {
     }
 }
 
-#[derive(Serialize, Deserialize, Clone, Debug, Default)]
-pub struct Mixer4chParams {
-    pub channels: [MixerChannelParams; 4]
+#[derive(Serialize, Deserialize, Clone, Debug)]
+pub struct MixerParams {
+    pub channels: Vec<MixerChannelParams>
+}
+
+impl MixerParams {
+    pub fn with_channels(n: usize) -> MixerParams {
+        MixerParams {
+            channels: vec![MixerChannelParams::default(); n]
+        }
+    }
+}
+
+impl Default for MixerParams {
+    fn default() -> Self {
+        Self::with_channels(4)
```

Comment:
Does `MixerParams` still need to impl `Default` now? I'm ok with a consequence of this refactor being that there's no default mixer anymore

Reply:
Yeah seems reasonable. I can pull it.

Reply:
Ah I remember my reason. It's just so `MixerParams` _has_ a default. It previously derived `Default` so I just wanted to preserve that interface. But yeah, I have no problem removing it either.

Reply:
Fixed in 32f240a